crates/ethereum/payload/src/filter.rs

```diff
@@ -0,0 +1,29 @@
+/// A filter that allows to check if a transaction satisfies a set of conditions
+pub trait TransactionFilter {
+    /// The type of the transaction to check.
+    type Transaction;
+
+    /// Returns true if the transaction satisfies the conditions.
+    fn is_valid(&self, transaction: &Self::Transaction) -> bool;
+}
+
+/// A no-op implementation of [`TransactionFilter`] which
+/// marks all transactions as valid.
+#[derive(Debug)]
+pub struct NoOpTransactionFilter<T>(std::marker::PhantomData<T>);
+
+impl<T> NoOpTransactionFilter<T> {
```

Comment:
I'd like to implement this for the unit type `()` which acts as the noop

Reply:
I'm not too sure how this would work with the AGT `Transaction` as `impl<T> TransactionFilter for ()` leaves `T` unconstrained. I could move `T` to `fn is_valid<Transaction>(&self, transaction: Transaction) -> bool;` but I am not sure this is what we want.

Otherwise I can also implement `TransactionFilter` for `PhantomData<T>`.